### ISSUE TITLE
Sale order as subscription does not allow combining different recurrences from website

### ISSUE THREAD

**Issue by fernandahf** [odoo/odoo#156843]
Impacted versions: 16.0 and 17.0

Steps to reproduce:

* Go to the backend and configure 2 products with different recurrences and publish them.
* Try to add both products to the cart.

You should see the following error in one of them:

"This product is not compatible with the product(s) in your cart."

Current behavior:

In version 14.0, when a user wants to purchase recurrent services, the combining of different recurrences was allowed, in 16.0, this feature was replaced with a very technical message on the website:

"This product is not compatible with the product(s) in your cart."

We do know the subscriptions in 16 are sale orders, and that is the main reason for the restriction, however what about the user experience on the website?

If the user wants to purchase different products, it does not matter what is the background configuration, Odoo must allow adding any product the user wants to purchase.

Expected behavior:

No warnings or errors trying to add products to the cart from the website.

Video/Screenshot link (optional):

https://youtu.be/7PgqJSP4VSY

Support ticket number submitted via odoo.com/help (optional):

OPW#3791727

**Comment by Yenthe666** [comment]
See also https://github.com/odoo/odoo/issues/118858 which contains an array of issues with the current subscription logic/flows in Odoo V16 and up :)

**Comment by codeagencybe** [comment]
Yeah this change is biting back hard now. 

I have shared many times already, the only proper way to handle this is moving the recurrence to sale order LINE level and splitting the sale order into multiple ones when the order is confirmed. Like this at least the quotation can handle the situation instead of breaking.

**Comment by Yenthe666** [comment]
Exactly. Without moving the recurrence to a line level it will never work out in real life usecases where a mixture of recurrence is very common. How this was initially built was a very wrong approach.. and in two versions nothing was learned.

**Comment by moylop260** [comment]
@mart-e 

Our customer is migrating from v14.0 to v16.0 but it is a blocking point for the project

They are looking for a meeting with a manager from their OPW's ticket:
 - https://www.odoo.com/my/tasks/3791727

But it was not replied yet

So, they want me to contact to Odoo's manager in order to know what should be the best solution for them

Do you know who could help us?

Thank you in advance!

**Comment by codeagencybe** [comment]
> @mart-e
> 
> Our customer is migrating from v14.0 to v16.0 but it is a blocking point for the project
> 
> They are looking for a meeting with a manager from their OPW's ticket:
> 
> * https://www.odoo.com/my/tasks/3791727
> 
> But it was not replied yet
> 
> So, they want me to contact to Odoo's manager in order to know what should be the best solution for them
> 
> Do you know who could help us?
> 
> Thank you in advance!

I already had a meeting with dalaOdoo (product owner) from subscriptions app last year and been requesting meetings with Odoo about how to go forward with this but they don't respond. 

The last update I got last year from Odoo was clear: it is what is and no changes happening to this. 
It seems they simply don't care about this broken state of efficiency :/
